libs/features/dynamic-navigation.js

```diff
@@ -0,0 +1,29 @@
+export default function dynamicNav(metadataContent, methods) {
+  const { getConfig, getMetadata } = methods;
+  const { dynamicNavKey } = getConfig();
+  const gnavSourceContent = getMetadata('gnav-source');
+
+  if (!dynamicNavKey) return;
+
+  if (metadataContent === 'entry' && gnavSourceContent) {
+    window.sessionStorage.setItem('gnavSource', gnavSourceContent);
+    window.sessionStorage.setItem('dynamicNavKey', dynamicNavKey);
+    return;
+  }
+
+  if (metadataContent === 'on') {
```

Comment:
If you stick to your early return ruleset, you can avoid the nested if/else
if (metadataContent !== 'on') return